pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -1305,24 +1315,87 @@ func TestDaemonKillFailedPods(t *testing.T) {
 		{numFailedPods: 0, numNormalPods: 0, expectedCreates: 1, expectedDeletes: 0, expectedEvents: 0, test: "no pods (create 1)"},
 		{numFailedPods: 1, numNormalPods: 0, expectedCreates: 0, expectedDeletes: 1, expectedEvents: 1, test: "1 failed pod (kill 1), 0 normal pod (create 0; will create in the next sync)"},
 		{numFailedPods: 1, numNormalPods: 3, expectedCreates: 0, expectedDeletes: 3, expectedEvents: 1, test: "1 failed pod (kill 1), 3 normal pods (kill 2)"},
-		{numFailedPods: 2, numNormalPods: 1, expectedCreates: 0, expectedDeletes: 2, expectedEvents: 2, test: "2 failed pods (kill 2), 1 normal pod"},
```

Comment:
Having 2 failed DS pods for 1 DS got rate limited by the new functionality

Reply:
does that make this test time-dependent? will that cause test flakes?

Reply:
I meant that before we were using a fake clock, if 50ms was exceeded during the test loop, this case might get hit. with a fake clock there should be no time-based flakiness